keras/src/models/model.py

```diff
@@ -556,6 +556,125 @@ def _get_variable_map(self):
         map_saveable_variables(self, store=store, visited_saveables=set())
         return store
 
+    def get_nested_variables(
```

Comment:
We should find a better name, maybe `get_state_tree()` (pytree?)

Reply:
Sounds good! I'll change it to `get_state_tree`.

Reply:
Good name!